editor/task_tree.cpp

```diff
@@ -305,27 +384,79 @@ bool TaskTree::_can_drop_data_fw(const Point2 &p_point, const Variant &p_data) c
 		return false;
 	}
 
-	if (!item->get_parent() && section != 0) { // before/after root item
+	if (!item->get_parent() && section != 0) { // Before/after root item.
 		return false;
 	}
 
 	if (String(d["type"]) == "task") {
-		Ref<BTTask> task = d["task"];
-		const Ref<BTTask> to_task = item->get_metadata(0);
-		if (task != to_task && !to_task->is_descendant_of(task)) {
-			return true;
+		TypedArray<BTTask> tasks = d["tasks"];
+		if (tasks.is_empty()) {
+			return false; // No tasks.
+		}
+		for (int i = 0; i < tasks.size(); i++) {
+			Ref<BTTask> task = tasks[i];
+			const Ref<BTTask> to_task = item->get_metadata(0);
+			if (to_task->is_descendant_of(task) || task == to_task ||
+					(task == to_task && task->get_index() + section >= to_task->get_index() && !item->is_collapsed() && item->get_child_count() > 0)) {
```

Comment:
This check is suspicious because it either won't be reached or evaluate to false, since we already check for `task == to_task` in the previous line

Reply:
I forgot to remove it. This section was to be replaced.